### PR TITLE
SDIT 927 mapping record change DTO and URL

### PR DESCRIPTION
**Pull request by andymarke**
This uses new endpoint that will eventually  create all mappings (hearings etc) in one go.

This means the DTO for create and read are now different
This means the base URL for create and read are now different (the base class assumed they were the same)

TODO - add hearing and punishment mappings to DTO

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/adjudications/AdjudicationsMappingService.kt [hunk 1]
```diff
@@ -9,48 +9,43 @@ import org.springframework.web.reactive.function.client.WebClientResponseExcepti
...
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomismappings.model.AdjudicationAllMappingDto
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomismappings.model.AdjudicationMappingDto
 
 @Service
 class AdjudicationsMappingService(@Qualifier("mappingApiWebClient") webClient: WebClient) :
-  MigrationMapping<AdjudicationMapping>(domainUrl = "/mapping/adjudications", webClient) {
+  MigrationMapping<AdjudicationAllMappingDto>(domainUrl = "/mapping/adjudications", webClient) {
 
-  suspend fun findNomisMapping(adjudicationNumber: Long, chargeSequence: Int): AdjudicationMapping? {
+  override fun createMappingUrl(): String {
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/adjudications/AdjudicationsMappingService.kt:19]: base class assumed the GET and POST urls were the same - in our case they are different,. We read the mapping for the main adjudication but write mapping for all child entities

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/adjudications/AdjudicationsMappingService.kt [hunk 2]
```diff
@@ -9,48 +9,43 @@ import org.springframework.web.reactive.function.client.WebClientResponseExcepti
...
       .retrieve()
       .bodyToMono(Unit::class.java)
-      .map { CreateMappingResult<AdjudicationMapping>() }
-      .awaitFirstOrDefault(CreateMappingResult<AdjudicationMapping>())
+      .map { CreateMappingResult<AdjudicationAllMappingDto>() }
+      .awaitFirstOrDefault(CreateMappingResult())
   }
 }
-
-data class AdjudicationMapping(
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/adjudications/AdjudicationsMappingService.kt:49]: now use generated classes

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/adjudications/AdjudicationsMigrationService.kt [hunk 3]
```diff
@@ -286,21 +290,15 @@ class AdjudicationsMigrationService(
...
 
-        val dpsAdjudication = adjudicationsService.createAdjudication(nomisAdjudication.toAdjudication())
-        val chargeNumber = dpsAdjudication.chargeNumberMapping.chargeNumber
-        createAdjudicationMapping(
-          adjudicationNumber = adjudicationNumber,
-          chargeSequence = chargeSequence,
-          chargeNumber = chargeNumber,
-          context = context,
-        )
+        val mapping = adjudicationsService.createAdjudication(nomisAdjudication.toAdjudication())
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/adjudications/AdjudicationsMigrationService.kt:293]: DPS actually more or less returns the mapping - which is nice of them :-)

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/adjudications/AdjudicationsMigrationService.kt [hunk 4]
```diff
@@ -310,36 +308,33 @@ class AdjudicationsMigrationService(
...
+    adjudicationId = AdjudicationMappingDto(
+      adjudicationNumber = mapping.chargeNumberMapping.oicIncidentId,
+      chargeSequence = mapping.chargeNumberMapping.offenceSequence.toInt(),
+      chargeNumber = mapping.chargeNumberMapping.chargeNumber,
+    ),
+    hearings = emptyList(),
+    punishments = emptyList(),
+    label = context.migrationId,
+    mappingType = MappingType.MIGRATED,
+  ).run {
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/adjudications/AdjudicationsMigrationService.kt:323]: will add hearings and punishments mapping in next PR

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**